tests/test_tarball.py

```diff
@@ -27,10 +28,15 @@ def test_packageurl(self, mock_build):
         """
         tarball.name = ''
         tarball.version = ''
+        tarball.giturl = ''
         tarball.url = url
         tarball.name_and_version('', '', FileManager())
-        self.assertEqual(tarball.name, name)
-        self.assertEqual(tarball.version, version)
+        self.assertEqual(name, tarball.name)
+        self.assertEqual(version, tarball.version)
+        if re.match("https?://github.com", url) != None:
+            self.assertIsNotNone(
+                    re.match("https://github.com/[^/]+/"+tarball.repo+".git",
+                    tarball.giturl))
```

Comment:
this is asserting that `re.match(...)` is not none, and displaying the `tarball.giturl`as the error message. Was this intended? Seems like we should be checking at least if `tarball.giturl` is not None.

Reply:
tarball.giturl is inside re.match(), so it's testing that string against the regex and checking that the match status is not None.
Nonetheless, I'll add self.assertIsNotNone(tarball.giturl) prior to that.

Reply:
ah, /me having `(())` dyslexia